### PR TITLE
Improve POC shutdown

### PR DESCRIPTION
**Pull request by chesterxgchen**
Fixes # .

### Description
The poc shutdown currently implemented as by touch shutdown.fl. This will trigger the FL server and FL Clients to kill the job process shutdown themselves. 

Although this works,  since its killing the child job process directly, user might see error messages such leak semaphore and other internal messages. 
This PR tried to avoid that by using Admin API to shutdown properly. The new steps are 
1) Instantiate the Flare API (session)
2) list the jobs and find the active running jobs 
3) if there are active running jobs, abort the jobs first, ignore the invalid job Ids 
4) calling session.api.do_command to shutdown -all  to trigger system shutdown (Client + Server) 
5) close the connection (locks) from Admin to Server
6) still using  shutdown.fl to trigger shutdown as double check (although seems unnecessary) 

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/lighter/poc_commands.py [hunk 1]
```diff
@@ -189,12 +192,102 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
     else:
         excluded.append(SC.FLARE_CONSOLE)
 
-    print(f"stop_poc at {poc_workspace}")
+    print(f"start shutdown NVFLARE")
     validate_poc_workspace(poc_workspace)
     gpu_ids: List[int] = []
+    admin_api = None
+    try:
+        admin_api = _get_admin_api(poc_workspace)
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:200]: I think we should change to use the new flare_api. It should be much easier.

**Review by yanchengnv — CHANGES_REQUESTED** [review]
Please change to use the new flare api.

File: nvflare/lighter/poc_commands.py [hunk 2]
```diff
@@ -189,12 +192,102 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+        overseer_agent=overseer_agent,
+        user_name="admin",
+        ca_cert="",
+        client_key="",
+        client_cert="",
+        poc=True
+    )
+    # Sleep three seconds to make sure we login to the system
+    import time
+    time.sleep(3)
```
Comment (yhwen) [inline nvflare/lighter/poc_commands.py:256]: Should use login_with_poc() to login, instead of sleep.

File: nvflare/lighter/poc_commands.py [hunk 3]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+    else:
+        return []
+
+
+def _get_admin_api(poc_workspace: str) -> FLAdminAPI:
+    config = _get_admin_config(poc_workspace)
+    admin_config = config["admin"]
+    upload_dir, download_dir = _get_transfer_dirs(admin_config, poc_workspace)
+    overseer_agent = _get_overseer_agent(admin_config)
+    admin_api = FLAdminAPI(
```
Comment (YuanTingHsieh) [inline nvflare/lighter/poc_commands.py:248]: If not using this method, we should remove it

File: nvflare/lighter/poc_commands.py [hunk 4]
```diff
@@ -21,6 +21,10 @@
 from typing import Dict, List, Optional
 
 from nvflare.cli_exception import CLIException
+from nvflare.fuel.flare_api.api_spec import JobNotFound, NoConnection
+from nvflare.fuel.flare_api.flare_api import new_secure_session
+from nvflare.fuel.hci.client.fl_admin_api import FLAdminAPI
```
Comment (YuanTingHsieh) [inline nvflare/lighter/poc_commands.py:26]: Some using new api don't need the old one

**Review by YuanTingHsieh — CHANGES_REQUESTED** [review]

File: nvflare/fuel/flare_api/flare_api.py [hunk 5]
```diff
@@ -437,19 +437,24 @@ def basic_cb_with_print(session: Session, job_id: str, job_meta, *cb_args, **cb_
     return True
 
 
-def new_secure_session(username: str, startup_kit_location: str, debug: bool = False, timeout: float = 10.0) -> Session:
+def new_secure_session(username: str,
+                       startup_kit_location: str,
+                       debug: bool = False,
+                       timeout: float = 10.0,
+                       secure_mode: bool = True) -> Session:
```
Comment (yanchengnv) [inline nvflare/fuel/flare_api/flare_api.py:444]: Why do we need secure_mode here? This method is intended for creating a secure session.
If you want insecure session, use new_insecure_session() method.

File: nvflare/lighter/poc_commands.py [hunk 6]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
-    print(f"stop_poc at {poc_workspace}")
+    print(f"start shutdown NVFLARE")
     validate_poc_workspace(poc_workspace)
     gpu_ids: List[int] = []
+    sess = None
+
+    try:
+        admin_dir = os.path.join(poc_workspace, "admin")
+        print("trying to connect to FL server")
+        sess = new_secure_session("admin", admin_dir, secure_mode=False)
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:204]: Change to: sess = new_insecure_session().

File: nvflare/lighter/poc_commands.py [hunk 7]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+
+def _get_running_job_ids(jobs: list) -> List[str]:
+    if len(jobs) > 0:
+        running_job_ids = [job for job in jobs if job["status"] == 'RUNNING']
+        return running_job_ids
+    else:
+        return []
+
+
+def _get_admin_api(poc_workspace: str) -> FLAdminAPI:
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:243]: Not needed

File: nvflare/lighter/poc_commands.py [hunk 8]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+        client_cert="",
+        poc=True
+    )
+    # Sleep three seconds to make sure we login to the system
+    import time
+    time.sleep(3)
+    return admin_api
+
+
+def _get_overseer_agent(admin_config):
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:264]: Not needed.

File: nvflare/lighter/poc_commands.py [hunk 9]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+    overseer_agent = instantiate_class(
+        class_path=admin_config["overseer_agent"]["path"],
+        init_params=admin_config["overseer_agent"]["args"],
+    )
+    return overseer_agent
+
+
+def _get_admin_config(poc_workspace):
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:272]: Not needed

File: nvflare/lighter/poc_commands.py [hunk 10]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+    admin_startup_dir = os.path.join(poc_workspace, "admin", "startup")
+    admin_config_file = os.path.join(admin_startup_dir, "fed_admin.json")
+    config = _load_admin_config(admin_config_file)
+    return config
+
+
+def _get_transfer_dirs(admin_config, poc_workspace):
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:279]: Not needed

File: nvflare/lighter/poc_commands.py [hunk 11]
```diff
@@ -189,12 +193,105 @@ def stop_poc(poc_workspace: str, excluded=None, white_list=None):
...
+    upload_path = admin_config.get("upload_dir")
+    download_path = admin_config.get("download_dir")
+    upload_dir = os.path.join(poc_workspace, "admin", upload_path)
+    download_dir = os.path.join(poc_workspace, "admin", download_path)
+    return upload_dir, download_dir
+
+
+def _load_admin_config(admin_config_path) -> dict:
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:287]: Not needed

File: nvflare/lighter/poc_commands.py [hunk 12]
```diff
@@ -21,6 +21,10 @@
 from typing import Dict, List, Optional
 
 from nvflare.cli_exception import CLIException
+from nvflare.fuel.flare_api.api_spec import JobNotFound, NoConnection
+from nvflare.fuel.flare_api.flare_api import new_secure_session
```
Comment (yanchengnv) [inline nvflare/lighter/poc_commands.py:25]: Change to import new_insecure_session

**Review by yanchengnv — CHANGES_REQUESTED** [review]
Please change to use new_insecure_session(); revert changes to flare_api for new_secure_session(); remove unused methods/functions.

**Review by yanchengnv — APPROVED** [review]
LGTM

**Comment by YuanTingHsieh** [comment]
Please fix unit test errors thanks

**Review by YuanTingHsieh — APPROVED** [review]

**Pull request merged**